test/unit/active_model/serializer/associations_test.rb

```diff
@@ -14,6 +14,36 @@ def test_associations_inheritance
         assert_equal([:comments],
                      another_inherited_serializer_klass._associations.keys)
       end
+      def test_multiple_nested_associations
+        parent = SelfReferencingUserParent.new(name: "The Parent")
+        child1 = SelfReferencingUser.new(name: "The child", parent: parent)
```

Comment:
Could you change `child1` to `child` please? It doesn't make much sense to have `1` since there is only one :wink: